### PR TITLE
feat(experiments): Allow editing experiment goal

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Part of #10426 

Refactors out `MetricSelector`, which is used to change secondary & main metric both.

Makes sure everything follows the same pattern, which makes it easier to debug / fix things when they go wrong.

Finally, adds a button to edit the main experiment metric when the experiment is running.

Fixes a small bug I found while doing this, where we didn't accept filters when group_type_index was 0

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/experiments/ExperimentPreview.tsx [hunk 1]
```diff
@@ -286,6 +299,13 @@ export function ExperimentPreview({
                                         ))}
                                     </Col>
                                 ))}
+                            {experiment?.start_date && (
+                                <div className="mb-2 mt-4">
+                                    <LemonButton type="secondary" onClick={openExperimentGoalModal}>
+                                        Change Experiment Goal
```
Comment (raquelmsmith) [inline frontend/src/scenes/experiments/ExperimentPreview.tsx:305]: Most buttons don't have all words in caps, should probably keep consistent.
```suggestion
                                        Change experiment goal
```

File: frontend/src/scenes/experiments/ExperimentPreview.tsx [hunk 2]
```diff
@@ -296,6 +316,56 @@ export function ExperimentPreview({
                     <ExperimentWorkflow />
                 </Col>
             )}
+            <LemonModal
+                isOpen={isExperimentGoalModalOpen}
+                onClose={closeExperimentGoalModal}
+                width={650}
+                title={'Change Experiment Goal'}
```
Comment (raquelmsmith) [inline frontend/src/scenes/experiments/ExperimentPreview.tsx:323]:
```suggestion
                title={'Change experiment goal'}
```

Same as above

**Review by raquelmsmith — APPROVED** [review]
Just a couple language suggestions, but otherwise 👍 This is exciting!

**Review by liyiy — APPROVED** [review]
Great start! I tested out the button flow and it works well, just didn't test to see whether the results of the goal changes were accurate though as I didn't spawn any local experiments data for it. 

I think some exciting future improvements we can make to this stuff is making the experiment goals more prominent and having more of a UX flow to it. Right now we don't have too much going on UX-wise in experiments but as we add more functionality and context it's important to highlight how the information on the screen can be digested so that the user experience isn't overwhelming 🤔

**Comment by neilkakkar** [comment]
Yep. I'd be very afraid of adding more data to the screen before knowing we really want it.

**Pull request merged**